src/home/fetch-pokemons.ts

```diff
@@ -0,0 +1,46 @@
+import {useDispatch} from 'react-redux'
+import {useCallback, useEffect, useState} from 'react'
+import {fetchPokemonsAction} from './redux/home.actions'
+
+export const useFetchPokemons = () => {
+    //React hook
+    const dispatch = useDispatch();
+    const [loading, setIsloading] = useState(false)
+    const [errored, setErrored] = useState(false)
+
+    const fetchPokemons = useCallback(async () => {
+        // assuming that the data wont change often we can improve the performance and save it into the local staorage
+        const pokemonListFromLocalStorage = localStorage.getItem('pokemonList')
```

Comment:
There is also a package that does something similar: https://github.com/rt2zz/redux-persist

Reply:
Also, it's not recommended (bad actually) to save big data on localStorage.